### PR TITLE
feat(checkout): show when a paid intro price steps up to the base price

### PR DESCRIPTION
**Pull request by rogersole**
## Summary

Stacked on #1032 — review/merge that first.

The RC Billing / Stripe pricing table only rendered a future-charge row for **free trials**. A subscription with a **paid intro phase** (1 week @ $1.49, then $9.90/month) showed `Total due today $1.49` and nothing else, so the customer was never told that $9.90 lands a week later.

Unlike the Paddle summary in #1032, this wasn't showing a *wrong* date — the row was simply absent. This adds it:

```
After 1 week, on Aug 14, 2026   $9.90
Total due today                 $1.49
```

Rendered only when there is an intro phase and no trial; the existing trial row already covers the trial + intro case.

The amount is the **base price**, not `priceBreakdown.totalAmountInMicros` — the latter is today's discounted intro total. This matches what the product header already shows for the intro case.

### The copy already existed

`ProductInfoPriceAfterIntroPrice` (`"After {{introPriceDuration}}, on {{renewalDate}}"`) was defined in `supportedLanguages.ts` and translated in every one of the 30+ locale files, but referenced nowhere in the code. This is the row it was written for — no new strings needed.

### Shared duration label

The intro-duration logic moves out of `pricing-summary.svelte` into `getIntroPriceDurationLabel` so both surfaces share it. Its English `"1 week"` → `"week"` collapse becomes an opt-in flag rather than implicit behaviour: it reads well in the header's *"First week for $1.49"* but not in *"After week, on Aug 14"*, so the table takes the `"After 1 week"` form. The header's rendering is unchanged.

### Not included

**Discounts** have the same missing row. Left out deliberately: no localization key exists for them, so it needs new copy across every locale, and it overlaps the known copy debt in `secure-checkout-rc.svelte` (`DISCOUNT_*_TERMS_INFO` are hardcoded English with a `TODO WEB-4205 - Translations`). Better as its own change alongside that work.

## Test plan

- [x] `pnpm test` — 780 passing
- [x] Typecheck / lint / `svelte-check` / build clean
- [x] New tests verified non-vacuous: stashing the `pricing-table.svelte` change makes the 2 behaviour tests fail while the 4 guards still pass
- [x] Header refactor covered by new `getIntroPriceDurationLabel` unit tests (both collapse modes, plus a non-English locale)
- [ ] Chromatic: two new `PricingTable` stories (intro price, trial + intro price)
- [ ] Manual: RC/Stripe checkout for a 1-week-intro monthly product shows both the after-intro row and the correct total due today

<details>
<summary>AI assistance</summary>

Investigated and implemented with Claude Code (Opus 5). Found during an audit of all three checkout surfaces prompted by the Paddle bug in #1032 — this gap was the quieter half of the same root cause, and the orphaned localization key suggests the row was intended at some point and got dropped.

🤖 Generated with [Claude Code](https://claude.com/claude-code)
</details>

---

> [!NOTE]
> **Low Risk**
> Checkout pricing display only; no payment or auth changes. Logic is gated on trial/discount presence with tests covering edge cases.
> 
> **Overview**
> The checkout **pricing table** now shows a future-charge row when a subscription has a **paid intro phase** and no free trial—e.g. *After 1 week, on Aug 14, 2026* with the **base** price ($9.90), while *Total due today* still reflects the intro amount ($1.49). It uses the existing `ProductInfoPriceAfterIntroPrice` localization key that was previously unused.
> 
> The row is **suppressed** when a trial is present (only the trial row is shown) or when a **discount** supersedes the intro phase, so the schedule shown matches what actually applies.
> 
> `pricing-summary` no longer builds intro-duration text inline; it uses shared `getPeriodDurationLabel` / `getOfferDuration`, with English singular collapse (`"week"` vs `"1 week"`) only for the header copy via an opt-in flag—the table keeps *After 1 week*.
> 
> `PricingTable` gains `introPricePhase` and `discountPhase` props; `product-info` passes them through; upgrade checkout passes nulls. Storybook adds intro-only and trial+intro stories; new component tests cover dates, multi-cycle intros, and guard cases.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit a03185ae542a5ad20397e9eedd3755e49d854685. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/stories/molecules/pricing-table.stories.svelte [hunk 1]
```diff
@@ -81,6 +83,23 @@
...
+/>
+<Story
+  name="Disabled Tax Trial And Intro Price"
+  args={{
+    trialPhase: subscriptionOptionWithTrial.trial,
+    introPricePhase:
+      subscriptionOptionWithSingleWeekIntroPriceRecurring.introPrice,
+    basePhase: subscriptionOptionWithSingleWeekIntroPriceRecurring.base,
+  }}
+/>
```
[the remark this answers is not in the document]
Comment (rogersole) [reply src/stories/molecules/pricing-table.stories.svelte:110]: Correct — fixed in 318af98. Both stories now pass a matching breakdown:

```js
priceBreakdown: getPriceBreakdownTaxDisabled(
  subscriptionOptionWithSingleWeekIntroPriceRecurring,
),
```

`getPriceFromOption` prefers `discount ?? introPrice ?? base`, so "Total due today" now snapshots at the intro $1.49 against the row's base $9.90, which is the step-up the PR adds. Previously both read $9.90.

File: src/ui/molecules/pricing-table.svelte [hunk 2]
```diff
@@ -57,6 +61,24 @@
...
+  // describe both step-ups. The trial row still states the trial end date, so
+  // that case is under-described rather than wrong.
+  const introDuration = $derived(
+    introPricePhase ? getOfferDuration(introPricePhase) : null,
+  );
+  const introPriceEndDate = $derived(
+    !trialEndDate && introDuration
+      ? getNextRenewalDate(new Date(), introDuration, true)
+      : null,
+  );
```
[the remark this answers is not in the document]
Comment (rogersole) [reply src/ui/molecules/pricing-table.svelte:86]: Correct — fixed in 942af1f. This is the mirror of [the same finding on #1032](https://github.com/RevenueCat/purchases-js/pull/1032#discussion_r3736989867): same class of bug, other surface. Good catch spotting it on both.

The row was gated only on `!trialEndDate`, so with a discount present it would have rendered an intro step-up that the discount schedule overrides.

`pricing-table` had no way to know: it receives `resolvedDiscount`, but that conflates purchase-option discounts with promo codes applied via `priceBreakdown.appliedDiscounts`, so it is the wrong signal to branch on. Threaded the real `discountPhase` through instead — `product-info.svelte` already computed it for `PricingSummary`, so it was one line there:

```js
const introPriceEndDate = $derived(
  !trialEndDate && !discountPhase && introDuration
    ? getNextRenewalDate(new Date(), introDuration, true)
    : null,
);
```

Regression test added (`is not rendered when a discount supersedes the intro price`), verified to fail against the previous commit. The discount stories now pass `discountPhase` too, so the story data matches what the component actually receives in the app.

Discounts still do not get a step-up row of their own — that needs new copy across every locale and is tracked as a follow-up alongside WEB-4205, as noted in the PR description.
